src/player.cpp

```diff
@@ -5063,61 +5065,266 @@ void player::suffer()
+
+                    add_msg( _( "%1$s says: \"%2$s\"" ), i_name, i_talk );
+                    done_effect = true;
+                }
+
+                // Skill raise
+                if( !done_effect && one_in( to_turns<int>( time_duration::from_hours( 12_hours ) ) ) ) {
+                    std::vector<std::pair<skill_id, int>> skills;
+                    for( auto &pair : *_skills ) {
+                        skills.emplace_back( std::make_pair( pair.first, pair.second.level() ) );
                     }
-                    break;
-                case 8:
-                    add_msg_if_player(m_bad, _("It's a good time to lie down and sleep."));
+                    std::pair<skill_id, int> i_skill = skills[rng( 0, skills.size() - 1 )];
+
+                    add_msg( m_good, _( "You increase %1$s to level %2$d" ), i_skill.first.c_str(), i_skill.second + 1 );
+                    done_effect = true;
+                }
+
+                // Talk to self
+                if( !done_effect && one_in( to_turns<int>( time_duration::from_hours( 4_hours ) ) ) ) {
+                    std::vector<std::string> talk_s{ "Hey, can you hear me?",
+                                                        "Don't touch me.",
+                                                        "What's your name?",
+                                                        "I thought you were my friend.",
+                                                        "How are you today?",
+                                                        "Shut up! Don't lie to me.",
+                                                        "Why would you do that?",
+                                                        "Please, don't go.",
+                                                        "Don't leave me alone!",
+                                                        "Yeah, sure.",
+                                                        "No way, man.",
+                                                        "Do you really think so?",
+                                                        "Is it really time for that?",
+                                                        "Sorry, I can't hear you.",
+                                                        "You've told me already.",
+                                                        "I know!",
+                                                        "Why are you following me?",
+                                                        "This place is dangerous, you shouldn't be here.",
+                                                        "What are you doing out here?",
+                                                        "That's not true, is it?",
+                                                        "Are you hurt?" };
+
+                    std::string i_talk_s = talk_s[rng( 0, talk_s.size() - 1 )];
+
+                    add_msg( _( "%1$s says: \"%2$s\"" ), name, i_talk_s );
```

Comment:
This `name` referes to the player characters name. Is this intended?

Reply:
It is self-talk.